### ISSUE TITLE
[Feature Request] RequiredScopeFilter is not support AllowAnonymousAttribute

### ISSUE THREAD

**Issue by jing8956** [AzureAD/microsoft-identity-web#1103]
**Is your feature request related to a problem? Please describe.**
I'm creating a controller. Most APIs required authorization and individual public.
Maybe it was my too young too simple :-(

**Describe the solution you'd like**
I think the following code is best way.

``` cs
[Authorize]
[RequiredScope("scope")]
[Route("api/[controller]")]
public class SampleController : ControllerBase
{
    [HttpGet("...")]
    public IActionResult AuthApi()
    {
        // ...
    }

    [<AllowAnonymous>]
    [<HttpGet("...")>]
    public IActionResult AnonymousApi()
    {
        // ...
    }
}
```
But return HTTP 500 and full stack info when call AnonymousApi. 
<details>
  <summary>response</summary>
  <pre><code>
System.UnauthorizedAccessException: IDW10204: The user is unauthenticated. The HttpContext does not contain any claims. 
   at Microsoft.Identity.Web.Resource.RequiredScopeFilter.ValidateEffectiveScopes(AuthorizationFilterContext context)
   at Microsoft.Identity.Web.Resource.RequiredScopeFilter.OnAuthorization(AuthorizationFilterContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

HEADERS
\=======
:authority: localhost:5001
:method: GET
:path: /api/sample
:scheme: https
Accept: application/json, text/plain, */*
Accept-Encoding: gzip, deflate, br
Accept-Language: zh-CN,zh;q=0.9,en;q=0.8,en-US;q=0.7,en-GB;q=0.6,ja;q=0.5
Host: localhost:5001
Referer: `https://localhost:5001/dashboard`
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36 Edg/89.0.774.63
sec-fetch-site: same-origin
sec-fetch-mode: cors
sec-fetch-dest: empty
  </code></pre>
</details>

**Describe alternatives you've considered**
Set RequiredScopeAttribute for method is a alternatives. But is dirty code.
<details>
  <summary>alternative</summary>

``` cs
[Authorize]
[Route("api/[controller]")]
public class SampleController : ControllerBase
{
    [RequiredScope("scope")]
    [HttpGet("...")]
    public IActionResult AuthApi1()
    {
        // ...
    }

    [RequiredScope("scope")]
    [HttpGet("...")]
    public IActionResult AuthApi2()
    {
        // ...
    }

    [RequiredScope("scope")]
    [HttpGet("...")]
    public IActionResult AuthApi3()
    {
        // ...
    }

    [<AllowAnonymous>]
    [<HttpGet("...")>]
    public IActionResult AnonymousApi()
    {
        // ...
    }
}
```

</details>

**Additional context**
Nothing

**Comment by jmprieur** [comment]
@jing8956 this is by design.
The filters (ASP.NET Core concept) are additive (the ones at the level of the class add-up to the ones at the level of the actions).

In your scenario you'd want to set the attribute on the protected actions, indeed.

I've seen many examples where developer had an anonymous controller, and protected controllers.

**Comment by jing8956** [comment]
@jmprieur My target is learning not business, so I try to keep my code clean and tidy.
I've looked all over [ASP.net Core authorization](https://docs.microsoft.com/aspnet/core/security/authorization/introduction?view=aspnetcore-5.0) 
I'm feel stranged that it barely talks about IAuthorizationFilter interface.
I found this [document](https://docs.microsoft.com/aspnet/core/mvc/controllers/filters?view=aspnetcore-5.0#authorization-filters) writes why ignore it:

> ... Prefer configuring the authorization policies or writing a custom authorization policy over writing a custom filter. ...
> ...
> Do **not** throw exceptions within authorization filters:

It's probably because this class library needs to be compatible with the .Net Framework so can't do it.

I'll choose to deprect this filter and write a custom authorization policy. 
Thank you for your help. :-)
